drivers/SmartThings/matter-window-covering/src/init.lua

```diff
@@ -137,75 +126,33 @@ end
 
 -- current lift percentage, changed to 100ths percent
 local function current_pos_handler(driver, device, ib, response)
-  local position = 0
-  if ib.data.value ~= nil then
-    position = 100 - math.floor((ib.data.value / 100))
-    device:emit_event_for_endpoint(
-      ib.endpoint_id, capabilities.windowShadeLevel.shadeLevel(position)
-    )
+  if ib.data.value == nil then
+    return
   end
-  if device:get_field(EVENT_STATE) == WindowCoveringEventEnum.OPERATIONAL_STATE_EVENT then
-    if not device:get_field(IS_MOVING) then
-      if position == 0 then
-        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.windowShade.windowShade.closed())
-      elseif position == 100 then
-        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.windowShade.windowShade.open())
-      elseif position > 0 and position < 100 then
-        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.windowShade.windowShade.partially_open())
-      else
-        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.windowShade.windowShade.unknown())
-      end
-      device:set_field(EVENT_STATE, WindowCoveringEventEnum.NO_EVENT)
-    end
+  local windowShade = capabilities.windowShade.windowShade
+  local position = 100 - math.floor((ib.data.value / 100))
+  device:emit_event_for_endpoint(ib.endpoint_id, capabilities.windowShadeLevel.shadeLevel(position))
+  if position == 0 then
```

Comment:
If we are no longer checking whether the shade is not moving, then what would happen if the shade is opening for example, and the Operational Status is received first, so the windowShade capability is set to "Opening". If the position is then received following this, then wouldn't the capability be updated to "Paritially Open", even if the shade is still moving?

Reply:
Are you mean following step?

1. Device send "Opening" value of Operation Status to client.
2. Device send position value to client during it's still moving.

In this case, client displays "Partially Open" for shade status even if the shade is still moving.
However, VDA is the only device that operate like that.
And now I updated sub driver that cover the VDA.

Reply:
Yes that's the situation I was referring to. I have seen that with the VDA as well, where it sends the updated position many times while moving. So, do the physical window covering devices you've tested with only send the position value after the shade is done moving?

Reply:
Yes, I never seen the devices that send the updated position many times while moving except VDA. and when such a device comes out, we can update the VID and PID to operate as a [sub driver](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1527/files#diff-59878b164008097e4769b5821bd5941f862ab2e47b551c0b7b60c7ba97828a25).

Reply:
The Nrf52840 window covering sample (Matter development kit) will update its position every 5% per time while moving, as I have previously tested.